### PR TITLE
Unify MYPYINDUCTOR and MYPY

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118469
* #118468
* #118467
* __->__ #118432
* #118418
* #118414

The original motivation for MYPYINDUCTOR was a faster type checking configuration that only checked a subset of files. With the removal of `follow_imports = ignore`, we are now able to use dmypy to do fast incremental typechecking, eliminating the need for this.

Perhaps erroneously, when I tee'ed up this PR I elected to delete the `follow_imports = skip` designations in the mypy-inductor.ini. This lead to a number of extra type error suppressions that I manually edited. You will need to review.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: mypy.ini [hunk 1]
```diff
@@ -270,7 +271,31 @@ ignore_missing_imports = True
...
 
 [mypy-torch._inductor.*]
-ignore_errors = True
+disallow_any_generics = True
 
 [mypy-torch._dynamo.*]
-ignore_errors = True
+disallow_any_generics = True
+
+[mypy-colorama.*]
```
Comment (Skylion007) [inline mypy.ini:279]: I missed this in my first approval, but colorama does actually have a `type-package`

Comment (ezyang) [reply mypy.ini:279]: OK, I'll try removing it in a subsequent PR

File: torch/_dynamo/test_case.py [hunk 2]
```diff
@@ -44,14 +44,14 @@ def run_tests(needs=()):
 class TestCase(TorchTestCase):
     @classmethod
     def tearDownClass(cls):
-        cls._exit_stack.close()
+        cls._exit_stack.close()  # type: ignore[attr-defined]
```
Comment (int3) [inline torch/_dynamo/test_case.py:47]: we should add `_exit_stack` as an uninitialized class member instead

i.e.

```
class TestCase(TorchTestCase):
   _exit_stack: <type>
```

File: torch/_inductor/fx_passes/serialized_patterns/central_index.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# mypy: ignore-errors
```
Comment (int3) [inline torch/_inductor/fx_passes/serialized_patterns/central_index.py:1]: this should probably be autogen'ed as part of `gen_attention_patterns.py`

File: torch/_inductor/fx_passes/mkldnn_fusion.py [hunk 4]
```diff
@@ -343,11 +343,11 @@ def _is_valid_binary(match, fn):
         if any(
             not (
                 hasattr(n.args[0], "meta")
-                and isinstance(n.args[0].meta.get("val", None), torch.Tensor)
+                and isinstance(n.args[0].meta.get("val", None), torch.Tensor)  # type: ignore[union-attr]
```
Comment (int3) [inline torch/_inductor/fx_passes/mkldnn_fusion.py:346]: I guess we should probably type `meta` as a Dict of Any values?

Comment (ezyang) [reply torch/_inductor/fx_passes/mkldnn_fusion.py:346]: SEND FIXES PLZ!!

**Pull request closed without merging**